packages/flame_forge2d/lib/forge2d_camera.dart

```diff
@@ -1,16 +1,36 @@
+import 'package:flame/components.dart';
+import 'package:flame/extensions.dart';
 import 'package:flame/game.dart';
 
+import 'flame_forge2d.dart';
+
 class Forge2DCamera extends Camera {
+  Body? _targetBody;
+
+  /// Used to minimize the amount of [Vector2] objects created
+  Vector2? _unprojectVector;
+  Vector2? _projectVector;
+
   @override
   Vector2 unprojectVector(Vector2 screenCoordinates) {
-    return position
-      ..add(screenCoordinates / zoom)
+    if (_unprojectVector == null) {
+      _unprojectVector ??= screenCoordinates.clone();
+    } else {
+      _unprojectVector?.setFrom(screenCoordinates);
+    }
+    return (_unprojectVector! / zoom)
+      ..add(position)
       ..y *= -1;
   }
 
   @override
   Vector2 projectVector(Vector2 worldCoordinates) {
-    return ((worldCoordinates.clone()..y *= -1) - position)..scale(zoom);
+    if (_projectVector == null) {
+      _projectVector ??= worldCoordinates.clone();
+    } else {
+      _projectVector?.setFrom(worldCoordinates);
+    }
+    return ((_projectVector!..y *= -1) - position)..scale(zoom);
```

Comment:
This also creates new vector every time

Reply:
It should, we shouldn't give back a the same vector object to several callers.